pipelines/package_publish.go

```diff
@@ -7,28 +7,38 @@ import (
 
 	"dagger.io/dagger"
 	"github.com/grafana/grafana-build/containers"
+	"github.com/grafana/grafana-build/slices"
 )
 
 // PublishPackage creates a package and publishes it to a Google Cloud Storage bucket.
 func PublishPackage(ctx context.Context, d *dagger.Client, src *dagger.Directory, args PipelineArgs) error {
-	packages, err := PackageFiles(ctx, d, src, args)
-	if err != nil {
-		return err
-	}
-
-	for distro, targz := range packages {
-		opts := TarFileOpts{
-			IsEnterprise: args.BuildEnterprise,
-			Version:      args.Version,
-			BuildID:      args.BuildID,
-			Distro:       distro,
+	// This bool slice stores the values of args.BuildEnterprise for each build
+	// --enterprise: []bool{true}
+	// --enterprise --grafana: []bool{true, false}
+	// if -- enterprise is not used it always returns []bool{false}
+	skipOss := args.BuildEnterprise && !args.BuildGrafana
+	isEnterpriseBuild := slices.Unique([]bool{args.BuildEnterprise, skipOss})
+	for _, isEnterprise := range isEnterpriseBuild {
+		args.BuildEnterprise = isEnterprise
```

Comment:
We can probably avoid some risk if we make a copy of args here instead of assigning to it directly.

```suggestion
	for _, isEnterprise := range isEnterpriseBuild {
		// Make a copy of args in this scope to avoid overwriting what was provided in the function args.
		args := args
		args.BuildEnterprise = isEnterprise
```

Reply:
Just pushe da change so the assignment to `args.BuildEnterprise` is not necessary.

Reply:
I think `PackageFiles` uses `args.IsEnterprise` to determine the package name. :thinking: maybe a small refactor is required here?